### PR TITLE
Add a test-env package for Substrate that runs a test chain in Docker

### LINKED ISSUE #14: Implement a test-env package for Substrate (similar to Tezos wrapper)
The current e2e test setup for the Substrate rpc-wrapper requires the tester to manually start an instance of the Substrate chain from the `examples` directory on a particular port. This is not documented and not well suited for running in CI

The Tezos wrapper implementation uses a test-env module which allows the testing code to start a docker image with a node ready to run tests against. It is proposed to use a similar approach for the example Substrate chain.

Things are trickier for Substrate since the chain is unique and we can't just use a pre-made docker image. In this case it will be required to use a container which builds and then hosts a dev node for the example chain.

This is [already created for the subtrate-node-template](https://github.com/substrate-developer-hub/substrate-node-template/blob/main/docker-compose.yml) and this can be used as is.

## Completion Criteria

- [x] A `substrate-polywrap-test-env` package has been added to the repo
- [x] Using this the test suite can manage its own blockchain instance and no external commands are needed to run the test suite for `rpc-wrapper`

### PR DESCRIPTION
**Pull request by willemolding**
Closes #14 

Following the Tezos example, this adds a test-env package that can be imported by the tests and exposes functions so start a local docker container running the Subtrate test chain (forum post chain)

e.g. 
```typescript
import { up, down } from "substrate-polywrap-test-env";

// This starts a docker container running the test chain
const response = await up();

// It also returns the endpoint for the node
console.log(response.node.url);

// kills the container
await down();
```

Changes:

- Adds test-env package
- Moves the Substrate example chain code to the test-env package
- Adds Dockerfile to build chain and create a new docker image with the build result
- Updates test in rpc-wrapper to import and use the test-env
- - This allows tests to be run by simply running `yarn test` in the rpc-wrapper package
---

Potential outstanding issues:
- The rpc-wrapper package depends on the test-env package via a local filesystem import. This is ok for now but will not work if the package is downloaded in isolation outside the repo. It was done this way as I don't know how publishing to NPM should work for this repo and who will have permissions. This should be fixed in the future

- To save time the test-env pulls a pre-build docker image from dockerhub for the test chain. Currently this needs to be published manually but in the future should be automatically build and deployed by CI

### CODE REVIEW

File: protocol/substrate/test-env/docker-compose.yml [hunk 1]
```diff
@@ -2,16 +2,17 @@ version: "3.2"
 
 services:
   dev:
-    container_name: node-template
-    image: paritytech/ci-linux:974ba3ac-20201006
+    container_name: substrate-polywrap-test-env
+    image: willemolding/substrate-polywrap-test-env:latest
```
Comment (willemolding) [inline protocol/substrate/test-env/docker-compose.yml:6]: I hosted the container for the moment just to get development. This is fine as long as no one has to modify the test chain..

This should really be on github container registry and deployed automatically by CI

File: protocol/substrate/rpc-wrapper/yarn.lock [hunk 2]
```diff
@@ -5410,6 +5415,11 @@ strip-json-comments@^3.1.1:
   resolved "https://registry.yarnpkg.com/strip-json-comments/-/strip-json-comments-3.1.1.tgz#31f1281b3832630434831c310c01cccda8cbe006"
   integrity sha512-6fPc+R4ihwqP6N/aIv2f1gMH8lOVtWQHoqC4yK6oSDVVocumAsfCqjkXnqiYMhmMwS/mEHLp7Vehlt3ql6lEig==
 
+"substrate-polywrap-test-env@file:../test-env":
```
Comment (willemolding) [inline protocol/substrate/rpc-wrapper/yarn.lock:5418]: I don't love this `file:..` dependency. Any thoughts on a better way to do this without setting up NPM deployments through CI?

Comment (clearloop) [reply protocol/substrate/rpc-wrapper/yarn.lock:5418]: we can do it with `yarn workspace`, I remember this is also part of the TODOs, I'll do it

File: protocol/substrate/rpc-wrapper/yarn-error.log [hunk 3]
```diff
@@ -0,0 +1,6022 @@
+Arguments:
```
Comment (clearloop) [inline protocol/substrate/rpc-wrapper/yarn-error.log]: please ignore this file

**Review by clearloop — APPROVED** [review]
please ignore this file `protocol/substrate/rpc-wrapper/yarn-error.log`

File: protocol/substrate/rpc-wrapper/yarn-error.log (hunk 3 above)
Comment (willemolding) [reply protocol/substrate/rpc-wrapper/yarn-error.log]: Oh no its back!! :P

File: protocol/substrate/rpc-wrapper/yarn.lock (hunk 2 above)
Comment (willemolding) [reply protocol/substrate/rpc-wrapper/yarn.lock:5418]: Sweet I'll leave that to you in a separate PR

Comment (willemolding) [reply protocol/substrate/rpc-wrapper/yarn.lock:5418]: See #17

**Comment by dOrgJelli** [comment]
Fantastic thank you for adding this @willemolding, it will fit nicely with the Polywrap CLI's `infra` command. More details here: https://docs.polywrap.io/quick-start/test-wasm-wrappers/infra-pipeline

**Pull request merged**